testing/opae-c/test_enum_c.cpp

```diff
@@ -0,0 +1,540 @@
+  EXPECT_EQ(
+      fpgaEnumerate(&filter, 1, tokens.data(), tokens.size(), &num_matches),
+      FPGA_OK);
+  EXPECT_EQ(num_matches, 1);
+
+  // afu num_errors
+  ASSERT_EQ(fpgaPropertiesSetNumErrors(filter, device.port_num_errors),
+            FPGA_OK);
+  EXPECT_EQ(
+      fpgaEnumerate(&filter, 1, tokens.data(), tokens.size(), &num_matches),
+      FPGA_OK);
+  EXPECT_EQ(num_matches, 1);
+
+  // invalid
+  ASSERT_EQ(fpgaPropertiesSetNumErrors(filter, invalid_device_.port_num_errors),
+            FPGA_OK);
+  EXPECT_EQ(
+      fpgaEnumerate(&filter, 1, tokens.data(), tokens.size(), &num_matches),
+      FPGA_OK);
+  EXPECT_EQ(num_matches, 0);
+}
+
+TEST_P(enum_c_p, guid) {
+  auto device = platform_.devices[0];
+  // fme guid
+  fpga_guid fme_guid, afu_guid, random_guid;
+  ASSERT_EQ(uuid_parse(device.fme_guid, fme_guid), 0);
+  ASSERT_EQ(uuid_parse(device.afu_guid, afu_guid), 0);
+  ASSERT_EQ(uuid_parse(invalid_device_.afu_guid, random_guid), 0);
+  ASSERT_EQ(fpgaPropertiesSetGUID(filter, fme_guid), FPGA_OK);
+  EXPECT_EQ(
+      fpgaEnumerate(&filter, 1, tokens.data(), tokens.size(), &num_matches),
+      FPGA_OK);
+  EXPECT_EQ(num_matches, 1);
+  // afu guid
+  ASSERT_EQ(fpgaPropertiesSetGUID(filter, afu_guid), FPGA_OK);
+  EXPECT_EQ(
+      fpgaEnumerate(&filter, 1, tokens.data(), tokens.size(), &num_matches),
+      FPGA_OK);
+  EXPECT_EQ(num_matches, 1);
+  // random guid
+  ASSERT_EQ(fpgaPropertiesSetGUID(filter, random_guid), FPGA_OK);
+  EXPECT_EQ(
+      fpgaEnumerate(&filter, 1, tokens.data(), tokens.size(), &num_matches),
+      FPGA_OK);
+  EXPECT_EQ(num_matches, 0);
+}
+
+TEST_P(enum_c_p, clone_token) {
+  EXPECT_EQ(
+      fpgaEnumerate(nullptr, 0, tokens.data(), tokens.size(), &num_matches),
+      FPGA_OK);
+  ASSERT_EQ(num_matches, 2);
+  fpga_token src = tokens[0];
+  fpga_token dst;
+  EXPECT_EQ(fpgaCloneToken(src, &dst), FPGA_OK);
+}
+
+TEST_P(enum_c_p, clone_wo_src_dst) {
+  EXPECT_EQ(
+      fpgaEnumerate(nullptr, 0, tokens.data(), tokens.size(), &num_matches),
+      FPGA_OK);
+  EXPECT_EQ(num_matches, 2);
+  fpga_token src = tokens[0];
+  fpga_token dst;
+  EXPECT_EQ(fpgaCloneToken(NULL, &dst), FPGA_INVALID_PARAM);
+  EXPECT_EQ(fpgaCloneToken(&src, NULL), FPGA_INVALID_PARAM);
+}
+
+TEST_P(enum_c_p, no_token_magic) {
+  EXPECT_EQ(
+      fpgaEnumerate(nullptr, 0, tokens.data(), tokens.size(), &num_matches),
+      FPGA_OK);
+  EXPECT_EQ(num_matches, 2);
+  fpga_token src,dst;
+  EXPECT_NE(fpgaCloneToken(&src, &dst), FPGA_OK);
+}
+
+
+TEST_P(enum_c_p, destroy_token) {
+  // null filter
+  uint32_t num_matches;
+  EXPECT_EQ(
+      fpgaEnumerate(nullptr, 0, tokens.data(), tokens.size(), &num_matches),
+      FPGA_OK);
+  ASSERT_EQ(num_matches, 2);
+  num_matches = 100;
+  for (auto t : tokens) {
+    if (t != nullptr) {
+      EXPECT_EQ(fpgaDestroyToken(&t), FPGA_OK);
+    }
+  }
+
+  EXPECT_EQ(fpgaDestroyToken(nullptr), FPGA_INVALID_PARAM);
+  _fpga_token *dummy = new _fpga_token;
```

Comment:
Doesn't _fpga_token live in xfpga?  Should we be testing this in the opae-c tests?